ui/app/components/oidc-auth-buttons/index.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  AuthMethod,
+  GetAuthMethodRequest,
+  GetOIDCAuthURLRequest,
+  ListOIDCAuthMethodsResponse,
+  Ref,
+} from 'waypoint-pb';
+
+import ApiService from 'waypoint/services/api';
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+
+interface OIDCAuthButtonsArgs {
+  model: ListOIDCAuthMethodsResponse.AsObject;
+}
+
+export default class OIDCAuthButtonsComponent extends Component<OIDCAuthButtonsArgs> {
+  @tracked model!: ListOIDCAuthMethodsResponse.AsObject;
+  @service api!: ApiService;
+
+  @task
+  async initializeOIDCFlow(authMethodProvider: AuthMethod.AsObject): Promise<void> {
+    let authMethodProviderName = authMethodProvider.name;
+    let urlRequest = new GetOIDCAuthURLRequest();
+    let authMethodrequest = new GetAuthMethodRequest();
+
+    let authMethodRef = new Ref.AuthMethod();
+    authMethodRef.setName(authMethodProviderName);
+    authMethodrequest.setAuthMethod(authMethodRef);
+    urlRequest.setAuthMethod(authMethodRef);
+    let randomArray = new Uint32Array(10);
+    window.crypto.getRandomValues(randomArray);
+    let nonce = randomArray.join('').slice(0, 20);
```

Comment:
Could we extract this out into a separate function? Maybe `generateNonce` or something like that. Just to make things a little clearer.

Reply:
Good point I was thinking about doing similar